src/app/[locale]/client-page.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import {useQuery} from '@tanstack/react-query';
+import {SearchResult} from '@/lib/dataset-fetcher';
+import DatasetCard from '@/components/dataset-card';
+import {Dataset} from '@/lib/dataset-fetcher';
+
+interface ClientPageProps {
```

Comment:
So long as we're not exporting this interface, I'd prefer the (generic) name `Props` (just like you did in `src/app/[locale]/layout.tsx`).